pkg/solana/txm/pendingtx.go

```diff
@@ -563,6 +576,52 @@ func (c *pendingTxContext) TrimFinalizedErroredTxs() int {
 	return len(expiredIDs)
 }
 
+func (c *pendingTxContext) IsTxReorged(sig solana.Signature, sigOnChainState txmutils.TxState) (string, bool) {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	// Grab in memory state of the signature
+	txInfo, exists := c.sigToTxInfo[sig]
+	if !exists {
+		return "", false
+	}
+
+	// Compare our in-memory state of the sig with the current on-chain state to determine if the sig had a regression
+	sigInMemoryState := txInfo.state
+	var hasReorg bool
+	switch sigInMemoryState {
+	case txmutils.Confirmed:
+		if sigOnChainState == txmutils.Processed || sigOnChainState == txmutils.Broadcasted || sigOnChainState == txmutils.NotFound {
```

Comment:
Wouldn't there be a case where the in-memory state differs from the on-chain state due to the chain being polled more recently, which wouldn't be a true reorg?

Reply:
There's no problem that we have different states as long as the on-chain state is greater or equal than our in-memory state. 

If this premise is false, it's because:
- A re-org occurred and we need to handle it 
- [Potential Lagging RPC edge case](https://github.com/smartcontractkit/chainlink-solana/pull/951#discussion_r1902131441) 
	- We are not sure if this is a real risk

Reply:
> https://github.com/smartcontractkit/chainlink-solana/pull/951#discussion_r1902131441
Yeah this is what I was getting at... I guess you guys have already thought it through though and nothing has come up in soak testing